routes/routes.go

```diff
@@ -15,28 +15,42 @@ const (
 	EmbeddedSuffix   = "/embed"
 	ResourceIdVarKey = "resource_id"
 	SlugVarKey       = "human_readable_slug"
+	CatchAllVarKey   = "uri"
 )
 
 // Clients - struct containing all the clients for the controller
 type Clients struct {
 	Storage storage.Provider
-	Api     stubs.InteractivesAPIClient
+	API     InteractivesAPIClient
 }
 
 // Setup registers routes for the service
-func Setup(cfg *config.Config, r *mux.Router, hc http.HandlerFunc, interactivesHandler http.HandlerFunc) {
+func Setup(_ *config.Config, r *mux.Router, hc http.HandlerFunc, interactivesHandler http.HandlerFunc) {
 	r.StrictSlash(true).Path(HealthEndpoint).HandlerFunc(hc)
-
-	// slug and resource_id (+ /embed)
-	r.StrictSlash(true).Path(getPath(false, true)).Methods(http.MethodGet).HandlerFunc(interactivesHandler)
-	r.StrictSlash(true).Path(getPath(true, true)).Methods(http.MethodGet).HandlerFunc(interactivesHandler)
-	// just resource_id (+ /embed)
-	r.StrictSlash(true).Path(getPath(false, false)).Methods(http.MethodGet).HandlerFunc(interactivesHandler)
-	r.StrictSlash(true).Path(getPath(true, false)).Methods(http.MethodGet).HandlerFunc(interactivesHandler)
-
-	if cfg.ServeFromEmbeddedContent {
-		r.StrictSlash(true).PathPrefix("/interactives").Methods(http.MethodGet).HandlerFunc(interactivesHandler)
-	}
+	// /interactives
```

Comment:
does the logic block full featured ? only embed allowed. If not think there is a ticket to address that.. can pick up later